### ISSUE TITLE
Media Picker - restrict on file type (.svg, .mp4 etc)

### ISSUE THREAD

**Issue by madsrasmussen** [umbraco/Umbraco-CMS#4921]
Sometimes you need to restrict a specific media picker to a particular type of file. It could be cases where you only want to allow SVG upload, video uploads, etc. A pre value field to specify which file types are allowed in the data type would be amazing.

I imagine the functionality could work very much like the "Allow items of type" in the Multinode tree picker.

---
_This item has been added to our backlog [AB#4622](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/4622)_

**Comment by bjarnef** [comment]
Maybe the "Allow items of type" should all be based on the media type similar to how MNTP use this for content type (document type).

It should probably be considered how it should work with the existing options `onlyImages` and `onlyFolders`. Does it still need these options if this feature is added or could it be re-designed a bit?
https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/mediapicker/mediapicker.controller.js#L32-L33

Would it be better to have a multi-picker to select the available media types instead of manual enter a comma delimited list of aliases? Maybe this multi-picker could also be used for document types and member types?

The other part of this would be to restrict to select media items where it attached media is of a specific type, e.g. .svg, .pdf, .mp4, .mp3, etc.

**Comment by hades200082** [comment]
I come across this requirement often when creating media pickers for video displays.

For example, I need a content type where the user must upload video files in multiple formats (mp4, webm & ogv) in order to create a HTML structure like this..

```html
 <video autoplay="" muted="" loop="" playsinline="" id="vid" preload="auto" poster="http://i.onionstatic.com/onion/1506/8/original/1200.jpg">
        <source src="https://s3.amazonaws.com/micada.org/public/Newnew.mp4" type="video/mp4">
        <source src="https://s3.amazonaws.com/micada.org/public/Newnew.webm" type="video/webm">
        <source src="https://s3.amazonaws.com/micada.org/public/Newnew.ogv" type="video/ogv">
    </video>
```

MP4 I make mandetory and have to other two as optional, only rendering the associated source if they are uploaded.

This means I need 3 separate media pickers, one specifically for each type.

**Comment by umbrabot** [comment]
Hiya @madsrasmussen,

Just wanted to let you know that we noticed that this issue got a bit stale and might not be relevant any more.

We will close this issue for now but we're happy to open it up again if you think it's still relevant (for example: it's a feature request that's not yet implemented, or it's a bug that's not yet been fixed). 

To open it this issue up again, you can write `@umbrabot still relevant` in  a new comment as the first line. It would be *super* helpful for us if on the next line you could let us know why you think it's still relevant.

For example:

>@umbrabot still relevant
>This bug can still be reproduced in version `x.y.z` 

This will reopen the issue in the next few hours.

Thanks, from your friendly Umbraco GitHub bot :robot: :slightly_smiling_face:

**Comment by FransdeJong** [comment]
@umbrabot still relevant

Ran into this requirement today. It would be great if we could somehow set only a pdf or a mp4 or whatever.

**Comment by bjarnef** [comment]
@madsrasmussen @nielslyngsoe I wonder if this is considered for the new Media Picker v3? Maybe also restrict by media type similar to MNTP, which would make the "Pick only images" and "Disable folder select" obsolete. https://github.com/umbraco/Umbraco-CMS/issues/8471

Furhermore "Minimum" and "Maximum" configuration properties would make it more consistent with Multi-URL Picker and MNTP and make it more flexible as the current configuration only allow a single or multiple - not e.g. restrict up to 5 or 10 items.

**Comment by kfr-dwarf** [comment]
Had the same request from a client today, would be great to see this feature in a future version. 
Minimum and maximum configuration properties would also be a great addition, as suggested by Bjarnef above.

**Comment by nul800sebastiaan** [comment]
Good news then! The new media picker in the upcoming 8.14.0 does exactly this, can configure what media types are allowed per media picker restrict it with a min/max number of items as well. 

![image](https://user-images.githubusercontent.com/304656/119976492-7226d500-bfb7-11eb-86b0-e39343fba96c.png)

And when you create an upload field on a media type you can restrict it to exactly the extensions you want!

![image](https://user-images.githubusercontent.com/304656/119977017-30e2f500-bfb8-11eb-9b6f-8a4f3a17459f.png)

So all of these feature requests are now covered and I'll close this one! 🙌

**Comment by Nikkelmann** [comment]
@nul800sebastiaan 
Having searched through the docs, issues, discussions and source code, I've decided against my better judgement to necro post on this issue - sorry - this is the most relevant spot I found.

Does the above mention implementation of MediaTypes enable limiting the Media Picker to specific file types, and if so, how?

In my specific use-case I want the user to only select images of type PNG.
